src/main/java/io/nats/client/impl/NatsMessageConsumer.java

```diff
@@ -19,9 +19,12 @@
 import java.io.IOException;
 
 class NatsMessageConsumer extends NatsMessageConsumerBase implements PullManagerObserver {
-    protected final PullRequestOptions rePullPro;
+    protected final ConsumeOptions opts;
     protected final int thresholdMessages;
     protected final long thresholdBytes;
+    protected final SimplifiedSubscriptionMaker subscriptionMaker;
+    protected final Dispatcher userDispatcher;
+    protected final MessageHandler userMessageHandler;
```

Comment:
making the PullRequestOptions on the fly, so need to keep the original user options, etc. around.